pymc/model/core.py

```diff
@@ -1396,11 +1422,16 @@ def create_value_var(
 
         # Make the value variable a transformed value variable,
         # if there's an applicable transform
-        if transform is UNSET:
+        if default_transform is UNSET:
             if rv_var.owner is None:
-                transform = None
+                default_transform = None
             else:
-                transform = _default_transform(rv_var.owner.op, rv_var)
+                default_transform = _default_transform(rv_var.owner.op, rv_var)
+
+        if transform is UNSET:
+            transform = default_transform
+        elif transform and default_transform:
```

Comment:
Be explicit about the possible cases at this point?
```suggestion
        elif transform is not None and default_transform is not None:
```